src/python/turicreate/toolkits/_internal_utils.py

```diff
@@ -99,7 +99,8 @@ def _find_only_column_of_type(sframe, target_type, type_name, col_name):
                 raise ToolkitError('No "{col_name}" column specified and more than one {type_name} column in "dataset". Can not infer correct {col_name} column.'.format(col_name=col_name, type_name=type_name))
             image_column_name = name
     if image_column_name is None:
-        raise ToolkitError('No %s column in "dataset".' % type_name)
+        raise ToolkitError('No "{col_name}" column specified and no column with expected type "{type_name}" is found.'.format(col_name=col_name, type_name=type_name)
```

Comment:
This looks like a fix for an unrelated bug (unrelated to "Documentation clarification for OD")

In general, you should avoid mixing together branches for different bugs. Each PR should just merge the changes/commits for the issue it's fixing!